src/modules/map/components/openlayers/OpenLayersGeoJSONLayer.vue

```diff
@@ -46,66 +46,58 @@ export default {
+                // filtering any feature that doesn't define an ID (otherwise it breaks the whole loading process in OL)
+                geoJsonData.features = geoJsonData.features.filter((feature) => feature.id)
 
                 // if the GeoJSON describes a CRS (projection) we grab it so that we can reproject on the fly if needed
-                const dataProjection = geojsonData.crs ? geojsonData.crs.properties.name : null
+                // according to the IETF reference, if nothing is said about the projection used, it should be WGS84
+                const dataProjection = geoJsonData.crs
+                    ? geoJsonData.crs.properties.name
+                    : CoordinateSystems.WGS84.epsg
 
-                // reprojecting the GeoJSON if not in EPSG:3857, the default projection for GeoJSON is WGS84
-                // as stated in the reference https://tools.ietf.org/html/rfc7946#section-4
-                // if another projection was set in the GeoJSON (through the "crs" property) we use it instead
-                let reprojectedGeoJSON
-                if (dataProjection) {
-                    if (dataProjection !== CoordinateSystems.WEBMERCATOR.epsg) {
-                        reprojectedGeoJSON = reproject(
-                            geojsonData,
-                            dataProjection,
-                            CoordinateSystems.WEBMERCATOR.epsg
-                        )
-                    } else {
-                        // it's already in the correct projection, we don't reproject
-                        reprojectedGeoJSON = geojsonData
-                    }
-                } else {
-                    // according to the IETF reference, if nothing is said about the projection used, it should be WGS84
-                    reprojectedGeoJSON = reproject(
-                        geojsonData,
-                        CoordinateSystems.WGS84.epsg,
-                        CoordinateSystems.WEBMERCATOR.epsg
-                    )
-                }
                 this.layer.setSource(
                     new VectorSource({
-                        features: new GeoJSON().readFeatures(reprojectedGeoJSON),
+                        features: new GeoJSON().readFeatures(geoJsonData, {
+                            dataProjection,
+                            featureProjection: getProjection(CoordinateSystems.WEBMERCATOR.epsg),
+                        }),
                     })
                 )
-                this.layer.setStyle(function (feature, res) {
-                    return style.getFeatureStyle(feature, res)
-                })
-            })
-            .catch((error) => {
-                log.error(
-                    'Error while fetching GeoJSON data/style for layer ' + this.layerId,
-                    error
-                )
-            })
+            } catch (err) {
+                log.error('Unable to load data for GeoJSON layer', this.layerId, err)
+            }
+        },
```

Comment:
Similar as above, might be a better idea to treat axios call and data parsing error differently ?